app/src/main/java/cellscanner/wowtor/github/com/cellscanner/MainActivity.java

```diff
@@ -59,24 +59,22 @@ protected void onCreate(Bundle savedInstanceState) {
             public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                 exportButton.setEnabled(!isChecked);
                 clearButton.setEnabled(!isChecked);
-                if (isChecked)
-                    startRecording();
-                else
-                    stopRecording();
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    if (isChecked)
+                        Recorder.startService(getApplicationContext());
+                    else
+                        Recorder.stopService(getApplicationContext());
+                } else {
+                    if (isChecked)
+                        startRecording();
+                    else
+                        stopRecording();
+
+                }
             }
         });
 
         Toast.makeText(getApplicationContext(), String.format("Cellscanner service is %srunning.", Recorder.inRecordingState(getApplicationContext()) ? "" : "not "), Toast.LENGTH_SHORT).show();
-
-        final Handler handler = new Handler();
-        Runnable timer = new Runnable() {
-            @Override
-            public void run() {
-                updateLogViewer();;
-                handler.postDelayed(this, 1000);
-            }
-        };
-        handler.post(timer);
```

Comment:
Dit gedeelte wordt ergens vervangen of gaat helemaal weg? Als je voorstelt dat het weg gaat, dan ook in layout.